clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsInstanceProvider.java

```diff
@@ -0,0 +1,171 @@
+package com.netflix.spinnaker.clouddriver.ecs.view;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.ec2.AmazonEC2;
+import com.amazonaws.services.ec2.model.DescribeInstanceStatusRequest;
+import com.amazonaws.services.ec2.model.InstanceStatus;
+import com.amazonaws.services.ecs.AmazonECS;
+import com.amazonaws.services.ecs.model.ContainerInstance;
+import com.amazonaws.services.ecs.model.DescribeContainerInstancesRequest;
+import com.amazonaws.services.ecs.model.DescribeTasksRequest;
+import com.amazonaws.services.ecs.model.ListClustersRequest;
+import com.amazonaws.services.ecs.model.ListClustersResult;
+import com.amazonaws.services.ecs.model.Task;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.EcsCloudProvider;
+import com.netflix.spinnaker.clouddriver.ecs.model.EcsInstance;
+import com.netflix.spinnaker.clouddriver.model.InstanceProvider;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Component
+public class EcsInstanceProvider implements InstanceProvider<EcsInstance> {
```

Comment:
By the way, here's what the https://spinnaker.prod.lkt.is:8084/instances/production/us-west-2/i-0b9bca7615be40e14 endpoint returned:


```{
  "amiLaunchIndex": 0,
  "architecture": "x86_64",
  "blockDeviceMappings": [
    {
      "deviceName": "\/dev\/sda1",
      "ebs": {
        "attachTime": 1503620510000,
        "deleteOnTermination": true,
        "status": "attached",
        "volumeId": "vol-0dbfb7b23b356d9c2"
      }
    }
  ],
  "clientToken": "4e7038fa-f878-4c6b-aa30-5229f8f66567_subnet-78d66d20_1",
  "cloudProvider": "aws",
  "cluster": "armoryspinnaker-ha",
  "ebsOptimized": false,
  "health": [
    {
      "healthClass": "platform",
      "state": "Unknown",
      "type": "Amazon"
    },
    {
      "instanceId": "i-0b9bca7615be40e14",
      "loadBalancers": [
        {
          "description": "N\/A",
          "healthState": "Up",
          "instanceId": "i-0b9bca7615be40e14",
          "loadBalancerName": "armoryspinnaker-ha-internal",
          "loadBalancerType": "classic",
          "reasonCode": "N\/A",
          "state": "InService"
        }
      ],
      "state": "Up",
      "type": "LoadBalancer"
    }
  ],
  "healthState": "Up",
  "hypervisor": "xen",
  "iamInstanceProfile": {
    "arn": "arn:aws:iam::022103483154:instance-profile\/SpinnakerInstanceProfile",
    "id": "AIPAJRF777LERM5GQDDWK"
  },
  "imageId": "ami-0bd23b73",
  "insightActions": [
    
  ],
  "instanceId": "i-0b9bca7615be40e14",
  "instanceType": "m3.2xlarge",
  "keyName": "lookout-spinnaker",
  "launchTime": 1503620510000,
  "monitoring": {
    "state": "enabled"
  },
  "name": "i-0b9bca7615be40e14",
  "networkInterfaces": [
    {
      "attachment": {
        "attachTime": 1503620510000,
        "attachmentId": "eni-attach-f2a79b17",
        "deleteOnTermination": true,
        "deviceIndex": 0,
        "status": "attached"
      },
      "description": "",
      "groups": [
        {
          "groupId": "sg-1c44ee64",
          "groupName": "armoryspinnaker-default"
        }
      ],
      "ipv6Addresses": [
        
      ],
      "macAddress": "0a:5d:a5:63:9c:d0",
      "networkInterfaceId": "eni-99e7309a",
      "ownerId": "022103483154",
      "privateDnsName": "ip-10-100-15-35.us-west-2.compute.internal",
      "privateIpAddress": "10.100.15.35",
      "privateIpAddresses": [
        {
          "primary": true,
          "privateDnsName": "ip-10-100-15-35.us-west-2.compute.internal",
          "privateIpAddress": "10.100.15.35"
        }
      ],
      "sourceDestCheck": true,
      "status": "in-use",
      "subnetId": "subnet-78d66d20",
      "vpcId": "vpc-4869b22c"
    }
  ],
  "placement": {
    "availabilityZone": "us-west-2c",
    "groupName": "",
    "tenancy": "default"
  },
  "privateDnsName": "ip-10-100-15-35.us-west-2.compute.internal",
  "privateIpAddress": "10.100.15.35",
  "productCodes": [
    
  ],
  "providerType": "aws",
  "publicDnsName": "",
  "rootDeviceName": "\/dev\/sda1",
  "rootDeviceType": "ebs",
  "securityGroups": [
    {
      "groupId": "sg-1c44ee64",
      "groupName": "armoryspinnaker-default"
    }
  ],
  "serverGroup": "armoryspinnaker-ha-v104",
  "sourceDestCheck": true,
  "state": {
    "code": 16,
    "name": "running"
  },
  "stateTransitionReason": "",
  "subnetId": "subnet-78d66d20",
  "tags": [
    {
      "key": "datadog-ignore",
      "value": "updates"
    },
    {
      "key": "datadog-stack",
      "value": "ha"
    },
    {
      "key": "aws:autoscaling:groupName",
      "value": "armoryspinnaker-ha-v104"
    },
    {
      "key": "name",
      "value": "armoryspinnaker-ha"
    },
    {
      "key": "product",
      "value": "cd"
    },
    {
      "key": "service",
      "value": "spinnaker"
    }
  ],
  "virtualizationType": "hvm",
  "vpcId": "vpc-4869b22c",
  "zone": "us-west-2c"
}```

That is a quite massive amount of ata :D

Reply:
It does indeed. Most of the data is AWS specific, we need to see what is worth returning on the ECS side and update this endpoint. The basics are covered